### PR TITLE
Fixing Edit Event functionality

### LINKED ISSUE #525: (Bug) Edit Event feature not functional
- **Issue Description:** 
The Edit Event option though existing is not functional and the code hasn't been linked to the API yet. Also there is a lot of copied code that would require refactoring. Also we need to stop users from editing events they didn't create.

- **Issue Severity:** Medium

- **Environment Details:** `editEventDialog.dart` 

- **Observed Behaviour:** On clicking `Edit this Event` the screen opens to edit event details but the edit functionality has not been implemented yet. There is no change in the Event details even after pressing the submit button.

- **Expected Behaviour:** Event Details should get edited as specified. Users should be able edit only those events which they created.

- **Steps to reproduce issue:** Open Events page -> Go to an event -> Click on the three dots -> Open Edit Event -> Enter Details -> Event Details not changed. 

- **Snapshots/Videos:** 

https://user-images.githubusercontent.com/56952811/112622591-43726e00-8e51-11eb-8aa7-3e5c7b0fa01f.mp4

- **Does this issue need immediate attention?** Yes.

- **Are you willing to work on this issue:** Yes. Kindly assign this to me @Sagar2366 @DeltaHarbinger

### PR DESCRIPTION
**Pull request by CoderMayhem**
**What kind of change does this PR introduce?**
This PR fixes the faulty edit event feature and makes it functional. 

**Did you add tests for your changes?**
No

**Summary**
Fixes #525

### CODE REVIEW

**Comment by Sagar2366** [comment]
@CoderMayhem seems like issue is still not fixed. 
Please share solution video too. Also, on saving updated event, bottom nav bar disables/remove.

**Comment by CoderMayhem** [comment]
Okay. I'll look into this.

**Comment by Sagar2366** [comment]
@CoderMayhem disable code formatter and send only changes required to fix this issue.

**Comment by CoderMayhem** [comment]
Sure, sorry for the trouble.

**Review by Sagar2366 — COMMENTED** [review]
> Sure, sorry for the trouble.

resolve conflicts on this PR.

File: lib/views/pages/login_signup/set_url_page.dart [hunk 1]
```diff
@@ -75,9 +75,9 @@ class _UrlPageState extends State<UrlPage> with TickerProviderStateMixin<UrlPage
   Future setApiUrl() async {
     setState(() {
       orgUrl =
-          "${dropdownValue.toLowerCase()}://${urlController.text}/talawa/graphql/";
+          "${dropdownValue.toLowerCase()}://${urlController.text}/graphql/";
```
[the remark this answers is not in the document]
Comment (CoderMayhem) [reply lib/views/pages/login_signup/set_url_page.dart]: Done

**Comment by CoderMayhem** [comment]
@Sagar2366 sir can you please review this at your convenience. I have removed most of the unnecessary format changes and fixed the delay problem. Right now my exams are going on so I am bit occupied. I'll open an issue for showing the toast (when user tries to edit an event he/she did not create) once my exams are over.

**Comment by Sagar2366** [comment]
@CoderMayhem seems like issue is still not fixed, please verify it again.
Editing event does not reflect changes on events page.

**Comment by CoderMayhem** [comment]
https://user-images.githubusercontent.com/56952811/114311943-72096d80-9b0e-11eb-9b10-b2beccd921ae.mp4

@Sagar2366 sir the events are getting edited,  **but you can edit only those which you have created**. So please try to create an event and then try editing it. I'll be updating the issue description and will add the feature to stop users from editing events they did not create in this PR itself.

**Comment by CoderMayhem** [comment]
@Sagar2366 the suggested changes have been made. Kindly review at your convenience.

**Review by Sagar2366 — APPROVED** [review]

**Pull request merged**